File: Pracka.Cup.API/Models/HistoryDto.cs
﻿namespace Pracka.Cup.API.Models
{
    using Pracka.Cup.Database.Enums;
    using System;

    public class HistoryDto
    {
        public int Id { get; set; }
        public TeamDto HomeTeam { get; set; }
        public PlayerDto PlayerHomeTeam { get; set; }
        public int GoalsHomeTeam { get; set; }

        public TeamDto AwayTeam { get; set; }
        public PlayerDto PlayerAwayTeam { get; set; }
        public int GoalsAwayTeam { get; set; }

        public GameTypeEnum GameType { get; set; }
        public DateTime GameDateUTC { get; set; }
        public DateTime CreatedUTC { get; set; }
        public DateTime ModifiedUTC { get; set; }
    }
}

File: Pracka.Cup.Web/src/api/models/history-dto.ts
import { GameType } from "./game-type-enum";
import { PlayerDto } from "./player-dto";
import { TeamDto } from "./team-dto";

export type HistoryDto = {
    id: number
    homeTeam: TeamDto
    playerHomeTeam: PlayerDto
    goalsHomeTeam: number

    awayTeam: TeamDto
    playerAwayTeam: PlayerDto
    goalsAwayTeam: number

    gameDateUTC: Date
    gameType: GameType
}